src/js/builtins/shell.ts

```diff
@@ -223,6 +310,7 @@ export function createBunShellTemplateFunction(ShellInterpreter) {
 
   BunShell[cwdSymbol] = defaultCwd;
   BunShell[envSymbol] = defaultEnv;
+  BunShell[throwsSymbol] = true;
 
   Object.defineProperties(BunShell, {
     Shell: {
```

Comment:
JS descriptors will default to `false` for unspecified descriptor properties like `configurable` and `writable` so if it's `configurable: false` you can exclude it and it will be treated as false. Since this is done at bootstrap (I'm guessing) we don't have to worry about prototype injection (or else we'd also want to `__proto__: null` the descriptors as JS reads them as inherited properties (not just own properties).

Reply:
Defining the descriptor properties _((value + writable for data properties, get + set for accessors) + configurable + enumerable)_ everywhere is an option in those cases (though a little verbose) where it's relevant. It also means if we miss a descriptor property we have a possible issue. The nice thing about a `__proto__: null,` approach is it's forgiving. This kind of vulnerability (prototype injection) effects other APIs too (DOM and JS)
* https://github.com/nodejs/node/pull/43270/files
* https://github.com/nodejs/node/pulls?q=is%3Apr+is%3Aclosed+prototype+pollution
* https://github.com/search?q=repo%3Anodejs%2Fnode+__proto__+null&type=pullrequests

 and is part of a larger issue (for example spreading values into array `[...a]` can be modified by changes to `Array.prototype[Symbol.iterator]` etc). 
 
I've seen folks wrestle with the pretty/ugly bit of the syntax _(I've been there too)_ but at some point security and maintainability > opinions about dangling underscores.